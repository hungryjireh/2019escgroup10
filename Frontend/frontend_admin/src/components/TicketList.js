import React from "react";
import TicketListItem from "./TicketListItem";
import "../style/TicketList.css";

const TicketList = ({ user, tickets }) => {
  const renderedTickets = tickets.map(ticket => {
    return <TicketListItem user={user} key={ticket.id} ticket={ticket} />;
  });

  return (
    <div className="ticketlist">
      <div className="ticketlist-header">
        <ul className="ticketlist-header-list">
          <li>Subject</li>
          <li>Status</li>
          <li>Requester</li>
          <li>Last Updated</li>
          <li>Group</li>
        </ul>
      </div>
      <div className="ticketlist-body">
        <ul className="ticketlist-body-list">{renderedTickets}</ul>
      </div>
    </div>
  );
};

export default {TicketList, renderedTickets};
