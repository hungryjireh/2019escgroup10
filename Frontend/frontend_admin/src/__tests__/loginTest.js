import React from "react";
import { render, cleanup, fireEvent, waitForElement, getByLabelText, getByText, getByTestId } from 'react-testing-library';
import SignInForm from "../components/SignIn/SignInForm";
import { fetchMock, FetchMock } from '@react-mock/fetch';
import TicketList from "../components/TicketList";

// import axiosMock from 'axios'

afterEach(cleanup);

const renderComponent = ({username, password}) => 
    render(
        username === 'admin' && password === 'happy' ?
        <FetchMock
        mocks={[
            { matcher: '/api/token/', method: 'POST', response: localStorage.setItem('jwt', 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTU0NzkzMDQxLCJqdGkiOiI2MWUxZDFiNzFkNTI0YTljOTAzMzdlNTk5NDc2OTExNiIsInVzZXJfaWQiOjF9._mdR8yW62IoF8PC88OKcdKe6kIXtfLGixfMV1l-h6ew') }
        ]}
        >
        <SignInForm />
        </FetchMock> :
        <FetchMock
        mocks={[
            { matcher: '/api/token/', method: 'POST', response: localStorage.setItem('jwt', null) }
        ]}
        >
        <SignInForm/>
        </FetchMock>
    );

test('Username field', async () => {
    const {getByText, container, debug} = render(<SignInForm />);

    const username = 'admin';
    const usernameField = container.querySelector('#username');
    fireEvent.change(usernameField, {target: {value: username}});
    expect(usernameField.value).toBe('admin');
})

test('Password field', async () => {
    const {getByText, container, debug} = render(<SignInForm />);

    const password = 'happy';
    const passwordField = container.querySelector('#password');
    fireEvent.change(passwordField, {target: {value: password}});
    expect(passwordField.value).toBe('happy');
})

test('Submit Button', async () => {
    const count = 0;
    const handleSubmit = ({count}) => {count : count + 1};
    const {getByText, container, debug} = render(<SignInForm onClick={handleSubmit({count})}/>);
    const submitButton = getByText('Sign In')
    fireEvent.click(submitButton);
    expect(count === 1);
})

test('Check jwt token for authenticated user', async () => {
    const {container, debug} = renderComponent({username: "admin", password: "happy"});
    expect(localStorage.getItem('jwt')).toBe('eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTU0NzkzMDQxLCJqdGkiOiI2MWUxZDFiNzFkNTI0YTljOTAzMzdlNTk5NDc2OTExNiIsInVzZXJfaWQiOjF9._mdR8yW62IoF8PC88OKcdKe6kIXtfLGixfMV1l-h6ew');
})

test('Check jwt token for unauthenticated user', async () => {
    const {container, debug} = renderComponent({username: "admin", password: "whoop"});
    expect(localStorage.getItem('jwt')).toBe("null");


})

test('Check for ticket list rendering', async () =>{
    const {getByTestId} = render (<TicketList />)
    expect(getByTestId("1")).toBeIntheDOM()
})

// test('Check for dashboard', async () => {
    
//     axiosMock.get.mockResolvedValueOnce({username: "admin" ,password: "happy"})
//     // const {asFragment} = renderComponent({username: "admin", password: "happy"});
//     const {container, getByText} = render(<Dashboard />)
//     expect(container.firstChild).toMatchSnapshot()

// })