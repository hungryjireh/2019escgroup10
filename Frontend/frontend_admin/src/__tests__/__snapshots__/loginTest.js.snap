// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for dashboard 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa bubYkx"
  >
    <h1
      class="sc-bwzfXH cprBNA"
    >
      Hello!
    </h1>
    <form
      class="sc-htpNat exKjCZ"
    >
      <label
        for="username"
      >
        Username
      </label>
      <input
        class="sc-bxivhb bboGsa"
        id="username"
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        class="sc-ifAKCX gAKSmR"
        id="password"
        type="password"
        value=""
      />
      <div
        class="warning sc-bZQynM ingVyq"
      />
      <button
        class="sc-EHOje iILhao"
        to="/dashboard"
      >
        Sign In
      </button>
    </form>
  </div>
</DocumentFragment>
`;
